trade-finance-manager-api/src/v1/services/deal-cancellation/deal-cancellation.service.ts

```diff
@@ -61,13 +61,7 @@ export class DealCancellationService {
   public static async submitDealCancellation({ dealId, cancellation, auditDetails }: SubmitDealCancellationParams) {
     console.info(`Submitting deal cancellation for ${dealId}`);
 
-    const { cancelledDeal, riskExpiredFacilityUkefIds } = await api.submitDealCancellation({ dealId, cancellation, auditDetails });
-
-    const ukefDealId = getUkefDealId(cancelledDeal.dealSnapshot);
-
-    if (!ukefDealId) {
-      throw new Error(`Failed to find deal Id on deal ${dealId} when submitting deal cancellation. No email has been sent`);
```

Comment:
Is it now impossible for this to happen?

Reply:
This should never occur (and was just added in as a type checking measure) since the deal is in TFM, it should have an ID. I can add this to `getTfmDealCancellationResponse` rather than using `as string` if you prefer?

Reply:
I'm wondering if we need it at all. Like you say, this service will only be called once a deal is in TFM - when it'll have a `ukefDealId`. Seems it would just be adding unnecessary noise.

Also, if it does end up having no `ukefDealId`, I presume it'd fail somewhere else and throw an error anyway.

Reply:
It wouldn't fail (on this endpoint call at least) & could send an email with `ukefDealId: undefined` - I think it might fail on other calls though & feels like such an edge case as to be ok?